### DISCUSSION TITLE
[Xlsx-Import] How to handle multiple rows of headers?

### DISCUSSION THREAD

**Discussion by fnick851** [Siemienik/XToolset#165] [Q&A] [answered]
Hi, how to read sheet with multiple rows of header? Is it possible at all? (For example, first row we have gender spanning two cells, second row we have male and female each in one cell, is it possible to preserve this data?) Thanks!

**Comment by Siemienik** [comment, marked as answer]
@fnick851 Thank you for asking me 👍  I will attempt to answer you as perfect as I can.

> Hi, how to read sheet with multiple rows of header? Is it possible at all? 

Yes it is by using `rowOffset: 2`


for instance:

```ts
    {
            type: 'list',
            worksheet: 'Person',
            columns: [
                {
                    index: 1, // column A
                    key: 'isMale',
                    mapper: (v: string) => v.toLowerCase() === 'x'
                },
                {
                    index: 2, // column B
                    key: 'isFemale',
                    mapper: (v: string) => v.toLowerCase() === 'x'
                },
                {index: 3, key: 'Name'},
            ],
            rowOffset: 2, // ignore first two header rows
        },
```

**Comment by fnick851** [comment]
@Siemienik  Thanks. Are all cells parsed as string by default without a mapper?

**Comment by Siemienik** [comment]
> @Siemienik Thanks. Are all cells parsed as string by default without a mapper?

Default mapper is defined by code `(v: string) => v`.

Source code:
https://github.com/Siemienik/xlsx-import/blob/be72568ed68b1922c81731e45beecd89c3283ce9/src/Importer.ts#L22
and:
https://github.com/Siemienik/xlsx-import/blob/be72568ed68b1922c81731e45beecd89c3283ce9/src/Importer.ts#L41

**Comment by fnick851** [comment]
@Siemienik Thank you!
